### ISSUE TITLE
Tray icons disappearing after update

### ISSUE THREAD

**Issue by bartolialberto** [digitalcreations/MaxTo#736]
**Describe the bug**

The subject says it all.

After the automatic update of this morning (2.2.0) the tray icons disappeared.

By modifying the application bar configuration (select/deselect "use small buttons"), the tray icons appear again. Then after a minute or so they disappear.

By terminating MaxTo tray icons are ok. By starting MaxTo they disappear immediately.

**System information:**
- Windows version:  Windows 10 professional (19941.1052)
- MaxTo version:  2.2.0

**Additional context**

**Comment by vegardlarsen** [comment]
@bartolialberto I have heard reports of this via e-mail, but never seen this in action. I suggest:

- reboot Windows
- remove any taskbars, such as the new "News and Interests" (which basically show the weather on my machine) that arrived in Windows 10.

Please let me know if either of those solve the problem.

**Comment by schkr** [comment]
Win 10 Build 21H1

**Comment by bartolialberto** [comment]
Unfortunately neither of those solves the problem.

I had already removed "news and interest"; reboot has not solved.

**Comment by Gappa** [comment]
Ah, so it is related to the MaxTo update :)

Temporary workaround: shutdown `explorer.exe` (via task manager) and start it again (just run `explorer.exe` from `File > Run new task`), that works for me (all the latest Win updates).

**Comment by vegardlarsen** [comment]
I have had one customer relay to me via e-mail that in his case this was connected to the program [Core Temp](https://www.alcpu.com/CoreTemp/), and that exiting that program solved his issue. I have not been able to verify this.

**Comment by schkr** [comment]
Confirmed new update works on Win20H2 after resetting to defaults. 21H2 is
broken on my end.

**Comment by bartolialberto** [comment]
With Win21H1 it seems to work.

Thanks.

**Comment by vegardlarsen** [comment]
I am re-opening this at this is still an issue we need to fix.

**Comment by schkr** [comment]
Were you able to duplicate, or need logs?

**Comment by vegardlarsen** [comment]
I am not able to duplicate this as of yet, and I believe logs will not be helpful to solve it in this case.

**Comment by bartolialberto** [comment]
The problem has come back

21H1 build 19043.1052

This is very annoying. I am simply unable to use MaxTo as I have to shut it down in order to see my tray icons.

**Comment by ahupp** [comment]
If my displayed tray icons are just the default set (e.g battery, wifi, volume, time, etc) then everything works fine.  If I run slack and leave in the "hidden" set of tray icons (behind the chevron) everything is also fine.  If I pull slack into the displayed set then they all disappear when running MaxTo.  Doesn't seem to be slack specific, also happens w/ Dropbox etc.

**Comment by schkr** [comment]
I'm also using DropBox and HW64. Good call out!

**Comment by bartolialberto** [comment]
After installing this optional Windows update everything seems to work fine:

[](https://support.microsoft.com/en-us/topic/june-21-2021-kb5003690-os-builds-19041-1081-19042-1081-and-19043-1081-preview-11a7581f-2a01-47d5-ba12-431709ee2248
)
The highlights of that update includes an item that seems pertinent:

Updates an issue that causes blurry text on the news and interests button on the Windows taskbar for some screen resolutions.

**Comment by bartolialberto** [comment]
After installing this optional Windows update everything seems to work fine:

[https://support.microsoft.com/en-us/topic/june-21-2021-kb5003690-os-builds-19041-1081-19042-1081-and-19043-1081-preview-11a7581f-2a01-47d5-ba12-431709ee2248](url)

The highlights of that update includes an item that seems pertinent:

_Updates an issue that causes blurry text on the news and interests button on the Windows taskbar for some screen resolutions._

**Comment by ahupp** [comment]
I applied KB5003690 as well as a pending Windows update and the problem is fixed for me.

**Comment by aphex** [comment]
Same for me with. Windows update with KB5003690 and all is normal again.

**Comment by vegardlarsen** [comment]
Thank you all for the updates, and the workaround. As I have not been able to reproduce this locally, KB5003690 will be the official solution for this issue.

**Comment by carlsheehaniris** [comment]
MaxTo 2.2.0

- I am having the same issue, none of the suggestions have made a difference. 
- I have uninstalled and installed a few legacy versions to see if that will make a difference with no luck.
- After boot there is no MaxTo icon in the tray rendering the program useless with no other way to access the MaxTo menu 

2021-07-12 09:30:18 [Server@2.2.0.764] [Information] [] Logger setup (online logging: True, process #9840)
2021-07-12 09:30:19 [Server@2.2.0.764] [Fatal] [] Unhandled exception "UnauthorizedAccessException" at "   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.Threading.EventWaitHandle..ctor(Boolean initialState, EventResetMode mode, String name, Boolean& createdNew, EventWaitHandleSecurity eventSecurity)
   at MaxTo.Core.EventWaitHandleFactory.For(String name)
   at MaxTo.Server.Program.<Main>d__0.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at MaxTo.Server.Program.<Main>(String[] args)"
System.UnauthorizedAccessException: Access to the path 'MaxTo-9840' is denied.
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.Threading.EventWaitHandle..ctor(Boolean initialState, EventResetMode mode, String name, Boolean& createdNew, EventWaitHandleSecurity eventSecurity)
   at MaxTo.Core.EventWaitHandleFactory.For(String name)
   at MaxTo.Server.Program.<Main>d__0.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at MaxTo.Server.Program.<Main>(String[] args)
2021-07-12 09:49:26 [Core@2.2.0.764] [Information] [] Logger setup (online logging: True, process #3296)
2021-07-12 09:49:27 [Core@2.2.0.764] [Information] [MaxTo.Core.Settings.DefaultConfigurationLocation] Found no specified configuration location file, using default settings location "C:\Users\carl.sheehan\AppData\Roaming\MaxTo"
2021-07-12 09:49:28 [Core@2.2.0.764] [Information] [MaxTo.Core.Settings.DefaultConfiguration] Setting update channel to Stable in default configuration
2021-07-12 09:49:28 [Core@2.2.0.764] [Information] [MaxTo.Core.Settings.DefaultConfiguration] Setting update channel to Stable in default configuration
2021-07-12 09:49:28 [Core@2.2.0.764] [Information] [] Logger setup (online logging: True, process #3296)
2021-07-12 09:49:28 [Core@2.2.0.764] [Information] [] Logger setup (online logging: True, process #3296)
2021-07-12 09:49:28 [Core@2.2.0.764] [Information] [MaxTo.Core.Shims.ChromiumHookBlockInitializer] Creating registry keys to unblock hooks for Chromium-based browsers: "Edge Dev, Edge Beta, Chromium, Google Chrome, Brave, Iridium".
2021-07-12 09:49:28 [Core@2.2.0.764] [Information] [] Read shims from "C:\Users\carl.sheehan\AppData\Local\MaxTo\shims.json"
2021-07-12 09:49:28 [Core@2.2.0.764] [Information] [MaxTo.Core.Update.UpdateService] Squirrel-installed app, so rewriting CLI batch file.
2021-07-12 09:49:28 [Core@2.2.0.764] [Information] [MaxTo.Core.Update.UpdateService] Checking for updates.
2021-07-12 09:49:28 [Core@2.2.0.764] [Information] [MaxTo.Core.Update.UpdateService] Found no new updates.
2021-07-12 09:49:28 [Core@2.2.0.764] [Information] [MaxTo.Core.Update.UpdateService] No updates found
2021-07-12 09:49:29 [Core@2.2.0.764] [Information] [MaxTo.Core.Settings.LoginRegistrationService] Could not find startup task, but MaxTo is set to start with Windows. Creating task.
2021-07-12 09:49:29 [Core@2.2.0.764] [Error] [MaxTo.Core.Settings.LoginRegistrationService] Could not register startup task during startup.
System.UnauthorizedAccessException: Access is denied. (Exception from HRESULT: 0x80070005 (E_ACCESSDENIED))
   at Microsoft.Win32.TaskScheduler.V2Interop.ITaskFolder.RegisterTaskDefinition(String Path, ITaskDefinition pDefinition, Int32 flags, Object UserId, Object password, TaskLogonType LogonType, Object sddl)
   at Microsoft.Win32.TaskScheduler.TaskFolder.RegisterTaskDefinition(String path, TaskDefinition definition, TaskCreation createType, String userId, String password, TaskLogonType logonType, String sddl)
   at Microsoft.Win32.TaskScheduler.TaskFolder.RegisterTaskDefinition(String path, TaskDefinition definition)
   at MaxTo.Core.Settings.LoginRegistrationService.CreateTask()
   at MaxTo.Core.Settings.LoginRegistrationService.Register()
   at MaxTo.Core.Settings.LoginRegistrationService.InitializeAsync()
2021-07-12 09:49:29 [Core@2.2.0.764] [Information] [] Was started as normal user, expected elevated. Restarting.
2021-07-12 09:49:29 [Core@2.2.0.764] [Information] [MaxTo.Core.RestartService] Restarting MaxTo with arguments "--elevated --waitforexit 3296"

**Comment by vegardlarsen** [comment]
Thank you for that log. Reading it I would expect the issue to go away after a reboot?

If not @carlsheehaniris, do you have any special restrictions (ACLs, policies or similar) on your computer?

**Comment by vegardlarsen** [comment]
I am closing this as it seems the problem has been resolved after the Windows update.
